posthog/models/property/util.py

```diff
@@ -769,16 +772,36 @@ def build_selector_regex(selector: Selector) -> str:
 
 def extract_tables_and_properties(props: List[Property]) -> TCounter[PropertyIdentifier]:
     counters: List[tuple] = []
+
+    class PropertyChecker(TraversingVisitor):
+        def __init__(self):
+            self.event_properties: List[str] = []
+            self.person_properties: List[str] = []
+
+        def visit_field(self, node: ast.Field):
+            if len(node.chain) > 1 and node.chain[0] == "properties":
+                self.event_properties.append(node.chain[1])
+
+            if len(node.chain) > 2 and node.chain[0] == "person" and node.chain[1] == "properties":
+                self.person_properties.append(node.chain[2])
+
+            if (
+                len(node.chain) > 3
+                and node.chain[0] == "pdi"
+                and node.chain[1] == "person"
+                and node.chain[2] == "properties"
```

Comment:
wait does this allow `pdi.person.properties.browser`?

Reply:
I'm confused as to why I'd do that

Reply:
That's the "behind the scenes" route to the field.

The events table currently looks like this:

```
class EventsTable(Table):
    uuid: StringDatabaseField = StringDatabaseField(name="uuid")
    event: StringDatabaseField = StringDatabaseField(name="event")
    properties: StringJSONDatabaseField = StringJSONDatabaseField(name="properties")
    timestamp: DateTimeDatabaseField = DateTimeDatabaseField(name="timestamp")
    team_id: IntegerDatabaseField = IntegerDatabaseField(name="team_id")
    distinct_id: StringDatabaseField = StringDatabaseField(name="distinct_id")
    elements_chain: StringDatabaseField = StringDatabaseField(name="elements_chain")
    created_at: DateTimeDatabaseField = DateTimeDatabaseField(name="created_at")

    # lazy table that adds a join to the persons table
    pdi: LazyTable = LazyTable(
        from_field="distinct_id", table=PersonDistinctIdTable(), join_function=join_with_max_person_distinct_id_table
    )
    # person fields on the event itself
    poe: EventsPersonSubTable = EventsPersonSubTable()

    # TODO: swap these between pdi and person_on_events as needed
    person: FieldTraverser = FieldTraverser(chain=["pdi", "person"])
    person_id: FieldTraverser = FieldTraverser(chain=["pdi", "person_id"])

    def clickhouse_table(self):
        return "events"
```

To get the person's properties we must do two joins: 1) get the `person_id` from `person_distinct_id2`, 2) get the `properties` from `person` using that `person_id`. 

Depending if Person-On-Events is enabled or not, you'll either be accessing `pdi.person.properties` or `poe.properties`. Usually behind the scenes by just typing `person.properties`, but perhaps not?

The bug here is that access for `poe.properties` is not also checked for... 😅 but I would have found that at the earliest when adding the switch that flips `person` between `pdi.person` and `poe`...